### PR TITLE
move generate prompt to utils

### PR DESCRIPTION
**Pull request by loklok-infi**
move generate_prompt relative functions to utils, so it could be reused.

### CODE REVIEW

**Review by tloen — COMMENTED** [review]
Thanks. Excited to restore some sanity to these scripts

File: finetune.py [hunk 1]
```diff
@@ -109,30 +87,7 @@ def tokenize(prompt):
...
-            f"""Below is an instruction that describes a task. Write a response that appropriately completes the request.
-
-### Instruction:
-{data_point["instruction"]}
-
-### Response:
-"""
-        )
-    )
+    user_prompt = (generate_prompt_by_data_point(data_point))
```
[the remark this answers is not in the document]
Comment (loklok-infi) [reply finetune.py]: Thanks, I added a parameter "omit_response" to fix it

**Comment by loklok-infi** [comment]
> I saw some localized version use translated versions of the template, as well as alternative templates with no prefix sentence for instance.
> 
> It would be great to use a json instead of hardcoding the template.
> 
> The french Vigogne for instance uses this template
> 
> ```
> PROMPT_DICT = {
>     "prompt_input": (
>         "Ci-dessous se trouve une instruction qui décrit une tâche, associée à une entrée qui fournit un contexte supplémentaire. Écrivez une réponse qui complète correctement la demande.\n\n"
>         "### Instruction:\n{instruction}\n\n### Entrée:\n{input}\n\n### Réponse:\n"
>     ),
>     "prompt_no_input": (
>         "Ci-dessous se trouve une instruction qui décrit une tâche. Écrivez une réponse qui complète correctement la demande.\n\n"
>         "### Instruction:\n{instruction}\n\n### Réponse:\n"
>     ),
> }
> ```
> 
> Original dict, from original Alpaca repo, is
> 
> ```
> PROMPT_DICT = {
>     "prompt_input": (
>         "Below is an instruction that describes a task, paired with an input that provides further context. "
>         "Write a response that appropriately completes the request.\n\n"
>         "### Instruction:\n{instruction}\n\n### Input:\n{input}\n\n### Response:"
>     ),
>     "prompt_no_input": (
>         "Below is an instruction that describes a task. "
>         "Write a response that appropriately completes the request.\n\n"
>         "### Instruction:\n{instruction}\n\n### Response:"
>     ),
> }
> ```
> 
> (note the missing \n after response)
> 
> In addition to this template, the string to split on `### Response:` is also to be specified, or infered from the prompt (last line from the template for instance, if not specified). I'd add a "response_split" key to that PROMPT_DICT for instance.
> 
> I can submit a PR for this if you'd like, so it's more generic.

sure, thank you

**Comment by tloen** [comment]
Subsumed into #184

**Pull request closed without merging**